packages/backend/src/app.ts

```diff
@@ -622,36 +632,37 @@ export class App {
         },
         validatorMiddlewareOptions
       ),
+      getWalletAddressUrlFromQuote,
       createTokenIntrospectionMiddleware({
         requestType: AccessType.Quote,
         requestAction: RequestAction.Read
       }),
       httpsigMiddleware,
+      getWalletAddressForSubresource,
       quoteRoutes.get
     )
 
     router.get(
       WALLET_ADDRESS_PATH + '/jwks.json',
-      createWalletAddressMiddleware(),
-      createValidatorMiddleware<WalletAddressKeysContext>(
+      createValidatorMiddleware(
         walletAddressServerSpec,
         {
           path: '/jwks.json',
           method: HttpMethod.GET
         },
         validatorMiddlewareOptions
       ),
-      async (ctx: WalletAddressKeysContext): Promise<void> =>
-        await walletAddressKeyRoutes.getKeysByWalletAddressId(ctx)
+      getWalletAddressUrlFromPath,
+      walletAddressKeyRoutes.get
     )
 
     // Add the wallet address query route last.
     // Otherwise it will be matched instead of other Open Payments endpoints.
     router.get(
       WALLET_ADDRESS_PATH,
-      createWalletAddressMiddleware(),
+      getWalletAddressUrlFromPath,
```

Comment:
Basically, if we are fetching the SPSP payment pointer, we don't need to call the validator middleware since its not Open Payments related